controllers/factory/scrapeconfig_build.go

```diff
@@ -0,0 +1,760 @@
+package factory
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	victoriametricsv1beta1 "github.com/VictoriaMetrics/operator/api/v1beta1"
+	"gopkg.in/yaml.v2"
+)
+
+func generateScrapeConfig(
```

Comment:
It'd be great to have a test for this function.

Reply:
added in latest commit, please check.